docs/DeveloperGuide.md

```diff
@@ -290,6 +211,74 @@ The diagram below details how the user's command to add a room propagates throug
 
 ![Adding a Room](images/room/AddRoomCommandSeqDiagram.png)
 
+### Resident-Room allocation feature
+
+The allocation feature is facilitated by `ResidentRoom` which is a pair value of
+valid `[Resident, Room]` that represents an existing resident allocated to a room.
+It should be able to support the following operations:
+* `ResidentRoom#getName()` - Returns the resident allocated for a given ResidentRoom.
+* `ResidentRoom#getRoomNumber()` - Returns the room assignment for a given ResidentRoom.
+
+A `ResidentRoomList` is a supplementary class that tracks all the `ResidentRoom` assignments. It should support
+the following operations.
+* `ResidentRoomList#add()` - Adds a `ResidentRoom` new allocation.
+* `ResidentRoomList#remove()` - Removes an existing `ResidentRoom` allocation.
+* `ResidentRoomList#contains()` - Checks if an allocation exists given the resident and room.
+
+There are two Resident-Room user allocation and deallocation commands, `alloc` and `dealloc` respectively.
+
+The `alloc` command will do the following:
+
+Example: `alloc ri/1 oi/2`
+* Check that the 1st resident exists.
+* Check that the 2nd room exists exists.
+* Check that the 1st resident has not already been allocated to the 2nd room.
+* Check that no other room is allocated to the 2nd resident.
+* Check that room the 2nd room is not occupied by any resident.
+
+  **If all the above is true,**
+* Set the `ROOM` of the resident to be the room number of the room.
+* Set the `OCCUPATION_STATUS` of the room to `Y`.
+
+  ![Activity Diagram of Allocation](images/residentroom/ResidentRoomAllocationActivityDiagram.png)
+
+The `dealloc` command will do the following:
+
+Example: `dealloc 1`
+* Check that the first resident exists.
+* Check that the first resident has been allocated to a room.
+
+  **If all the above is true,**
+* Set the `ROOM` of  the resident to be `Room unallocated`.
+* Set the `OCCUPATION_STATUS` of the room to `N`.
+
+![Sequence Diagram of Deallocation](images/residentroom/ResidentRoomDeallocationSeqDiagram.png)
```

Comment:
Maybe add a bit of text explaining what this diagram shows? When I was reading this section, it seemed to come out of nowhere.

Reply:
On a more general note, if `dealloc` isn't a very different command (in terms of execution flow) from regular ones, this sequence diagram may not be strictly necessary.